src/3d/qgs3dmapscene.cpp

```diff
@@ -247,6 +247,36 @@ void Qgs3DMapScene::viewZoomFull()
   mCameraController->resetView( 1.5 * std::sqrt( a * a - side * side ) );  // assuming FOV being 45 degrees
 }
 
+void Qgs3DMapScene::viewExtent( const QgsRectangle &extent )
+{
+  QgsPointXY center = extent.center();
+  QgsVector3D centerWrld = mMap.mapToWorldCoordinates( QVector3D( center.x(), center.y(), 0 ) );
+  QgsVector3D p1 = mMap.mapToWorldCoordinates( QVector3D( extent.xMinimum(), extent.yMinimum(), 0 ) );
+  QgsVector3D p2 = mMap.mapToWorldCoordinates( QVector3D( extent.xMaximum(), extent.yMaximum(), 0 ) );
+
+  float side = std::max( std::abs( p1.x() - p2.x() ), std::abs( p1.z() - p2.z() ) );
+  float a =  side / 2.0f / std::sin( qDegreesToRadians( cameraController()->camera()->fieldOfView() ) / 2.0f );
```

Comment:
"a" should be [hypotenuse](https://en.wikipedia.org/wiki/Hypotenuse) of the orthogonal triangle, right? typically it is marked as "c" which confused me a bit...

If I understand correctly, we first calculate hypotenuse and then use pythagoras theorem to get the remaining side. While that's fine (although I think you should be using `side*side/4` instead of `side*side`), there would be less calculation to do if you used tangent to get the distance directly, something like `float distance = side / 2 / tan( fov / 2 );`